Ship_Game/Universe/SolarBodies/Planet/Planet_EvaluateBuildings.cs

```diff
@@ -469,19 +476,28 @@ bool SuitableForBuild(Building b, float budget, IReadOnlyList<Building> building
             return budget > 0 && b.ActualMaintenance(this) <= budget;
         }
 
-        bool SuitableForScrap(Building b, bool replacing)
+        bool SuitableForScrap(Building b, bool overBudget, float storageInUse, bool scrapZeroMaintenance, bool replacing)
         {
             if (b.IsBiospheres
-                || !b.Scrappable
-                || b.IsPlayerAdded && OwnerIsPlayer
-                || b.IsTerraformer
                 || b.IsMilitary
-                || b.BuildOnlyOnce
+                || !b.Scrappable
+                || b.IsSpacePort && Owner.GetPlanets().Count == 1 // Dont scrap our last spaceport
+                || b.BuildOnlyOnce)
+            {
+                return false;
+            }
+
+            if (!RequiredInBlueprints(b))
+                return true;
+            else if (!overBudget)
+                return false;
+
+            if (b.IsPlayerAdded && OwnerIsPlayer
                 || b.MoneyBuildingAndProfitable(b.ActualMaintenance(this), PopulationBillion)
                 || !WillMaintainPositiveFoodOutput(b)
-                || HasBlueprints && !Blueprints.BuildingSuitableForScrap(b)
-                || b.IsSpacePort && Owner.GetPlanets().Count == 1 // Dont scrap our last spaceport
-                || !IsBuildingOnHabitableTile(b) && replacing)  // Dont allow buildings on non habitable tiles to be scrapped when replacing
+                || !IsBuildingOnHabitableTile(b) && replacing  // Dont allow buildings on non habitable tiles to be scrapped when replacing
```

Comment:
maybe  add brackets to make it easier for others

Reply:
Brackets are not really needed. I usually use a line per logical phrase. (&& is higher priority than ||)